src/main.ts

```diff
@@ -0,0 +1,118 @@
+import * as core from '@actions/core'
+import * as util from 'util'
+import { ExecOptions } from '@actions/exec/lib/interfaces';
+import { ToolRunner } from '@actions/exec/lib/toolrunner';
+import { getTrivy } from './trivyHelper'
+import { getDockle } from './dockleHelper'
+const download = require('download');
+
+let trivyEnv: { [key: string]: string } = {};
+let dockleEnv: { [key: string]: string } = {};
+
+async function getWhitelistFileLoc(whitelistFilePath: string, whitelistFileBranch: string): Promise<string> {
+    const whitelistFileUrl = `https://raw.githubusercontent.com/${process.env.GITHUB_REPOSITORY}/${whitelistFileBranch}/${whitelistFilePath}`;
+    const whitelistFilePathParts = whitelistFilePath.split('/');
+    const whitelistFileName = whitelistFilePathParts[whitelistFilePathParts.length - 1];
+    const whitelistFileDownloadDir = `${process.env['GITHUB_WORKSPACE']}/_temp/containerScanWhitelist`;
+    const githubToken = core.getInput("github-token");
+    console.log(util.format("Downloading whitelist file from %s", whitelistFileUrl));
+
+    return download(whitelistFileUrl, whitelistFileDownloadDir, { headers: { Authorization: `token ${githubToken}` } }).then(() => {
+        return `${whitelistFileDownloadDir}/${whitelistFileName}`;
+    });
+}
+
+async function setTrivyEnvVariables() {
+    for (let key in process.env) {
+        trivyEnv[key] = process.env[key] || "";
+    }
+
+    const username = core.getInput("username");
+    const password = core.getInput("password");
+    if (username && password) {
+        trivyEnv["TRIVY_AUTH_URL"] = "https://registry.hub.docker.com";
```

Comment:
Should figure this out from image name (for other registries).

Reply:
The [documentation](https://github.com/aquasecurity/trivy#amazon-ecr-elastic-container-registry) is a bit confusing. Different ways to authenticate to different registries! We should test out other registries (ACR).

Reply:
Created a separate file for all the inputs.